clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -259,6 +260,25 @@ public void testCloseAdminClient() {
         }
     }
 
+    /**
+     * Test if admin client can be closed in the callback invoked when
+     * an api call completes.
+     *
+     * KAFKA-9330: If calling close in callback, AdminClient thread hangs
+     */
+    @Test(timeout = 5_000)
+    public void testCloseAdminClientInCallback() throws InterruptedException {
+        AdminClientUnitTestEnv env = mockClientEnv();
```

Comment:
Can we change this to use MockTime instead?

Reply:
Done.